src/main/java/com/taboola/backstage/Backstage.java

```diff
@@ -294,7 +312,22 @@ public Backstage build() {
             RestAPIClient backstageClient = restAPIClientBuilder.setBaseUrl(baseUrl).setUserAgentPrefix("Backstage").build();
             RestAPIClient authenticationClient = restAPIClientBuilder.setBaseUrl(authBaseUrl).setUserAgentPrefix("Authentication").build();
 
-            BackstageEndpointsFactory endpointsFactory = new BackstageEndpointsRetrofitFactory(backstageClient, authenticationClient);
+            RestApiTrackingClient.RestAPIClientBuilder restAPITrackClientBuilder = RestApiTrackingClient.builder()
+                    .setObjectMapper(objectMapper)
+                    .setConnectionTimeoutMillis(connectionTimeoutMillis)
+                    .setMaxIdleConnections(maxIdleConnections)
+                    .setReadTimeoutMillis(readTimeoutMillis)
+                    .setWriteTimeoutMillis(writeTimeoutMillis)
+                    .setSerializationConfig(serializationConfig)
+                    .setKeepAliveDurationMillis(keepAliveDurationMillis)
+                    .setExceptionFactory(new BackstageAPIExceptionFactory(objectMapper))
+                    .setCommunicationInterceptor(communicationInterceptor)
+                    .setDebug(debug);
+
+            RestApiTrackingClient trackClient = restAPITrackClientBuilder.setBaseUrl(trackingBaseUrl).build();
```

Comment:
cannot we use the RestApiClientBuilder that already exists?

Reply:
Unfortunately not, because it is from another lib/repo and it has some code to inject a special header that we should not do while tracking conversion.

Reply:
ok.